### PR TITLE
Add support for Google Pay

### PR DESCRIPTION
**Pull request by thomas-gr4vy**

### CODE REVIEW

File: packages/embed/src/popup/popup.ts [hunk 1]
```diff
@@ -22,14 +24,16 @@ export const createPopupController = (
   })
 
   subject.approvalLost$.subscribe(() => {
-    popup.current.stopCallback()
-    popup.current.popup.close()
-    subject.approvalStarted$.next()
+    if (popup.current) {
```
Comment (douglaseggleton) [inline packages/embed/src/popup/popup.ts:27]: `approvalLost$` should not fire from the overlay as their is no button to relaunch the popup?
This change makes sense though.

File: packages/embed/src/overlay/overlay.ts [hunk 2]
```diff
@@ -77,7 +77,10 @@ export const createOverlayController = (
 
   // assume an approval url with pre-defined overlay is an iframe
   subject.approvalUrl$.subscribe((url) => {
-    if (!subject.mode$.value()?.overlay) {
+    if (
+      !subject.mode$.value()?.overlay ||
+      (subject.mode$.value()?.overlay && !subject.mode$.value()?.popup)
+    ) {
```
Comment (douglaseggleton) [inline packages/embed/src/overlay/overlay.ts:83]: I'm not sure this makes sense. Maybe a separate `showOverlay$`?

File: packages/embed/src/subjects.ts [hunk 3]
```diff
@@ -45,13 +45,13 @@ export const createSubjectManager = () => {
   }
 
   subjects.formSubmit$.subscribe(() => {
-    if (subjects.mode$.value()?.popup) {
+    if (subjects.mode$.value()?.popup || subjects.mode$.value()?.overlay) {
```
Comment (douglaseggleton) [inline packages/embed/src/subjects.ts:48]: I think we should try and detatch popup from overlay. Something like this?

```
if (subjects.mode$.value()?.overlay) {
  subjects.showOverlay$.next()
}

```

File: packages/embed/src/popup/popup.ts (hunk 1 above)
Comment (thomas-gr4vy) [reply packages/embed/src/popup/popup.ts:27]: In the case of GPay, no, there is no reopen button. 👍

**Comment by douglaseggleton** [comment]
Looks good - Just need to label it up for release.

**Review by douglaseggleton — APPROVED** [review]
Looks good - Just need to label it up for release.

**Comment by gr4vy-code** [comment]
:rocket: PR was released in [`v2.8.0`](https://github.com/gr4vy/gr4vy-embed/releases/tag/v2.8.0) :rocket:

**Pull request merged**